flow/environments/xsede.py

```diff
@@ -66,6 +68,41 @@ def add_args(cls, parser):
                   'If omitted, uses the system default '
                   '(slurm default is "slurm-%%j.out").'))
 
+    @classmethod
+    def get_prefix(cls, operation, mpi_prefix=None, cmd_prefix=None, parallel=False):
+        """Template filter for getting prefix based on environment and proper directives.
+
+        :param operation:
+            The operation for which to add prefix.
+        :param mpi_prefix:
+            User defined mpi_prefix string. Default is set to None.
+        :param cmd_prefix:
+            User defined cmd_prefix string. Default is set to None.
+        :param parallel:
+            If True, operations are assumed to be executed in parallel, which means
+            that the number of total tasks is the sum of all tasks instead of the
+            maximum number of tasks. Default is set to False.
+        :return prefix:
+            The prefix should be added for the operation.
+        :type prefix:
+            str
+        """
+        prefix = ''
+        if operation.directives.get('omp_num_threads'):
+            prefix += 'export OMP_NUM_THREADS={}\n'.format(operation.directives['omp_num_threads'])
+        if mpi_prefix:
+            prefix += mpi_prefix
+        elif operation.directives.get('nranks'):
+            if parallel:
```

Comment:
Is there a way that we could find the offset's ourselves by keeping track of all the requested ranks, and dividing them correctly? We could have a policy to fail if enough ranks aren't requested or that we give the extra ranks to the first operation or something like that.

Reply:
My personal thought would be to raise an error, but the offset could be managed automatically regardless by keeping track of the ranks assigned and managing offset according.

Reply:
I'd prefer to generally not change the behavior as part of this patch. So we should just try to emulate the behavior of the template implementation as closely as possible (ideally exactly).

Reply:
Okay, that is fine then. I will resolve the conversation, but I think that would be useful to have in the future when dealing with submitting using run. Having the user specify offsets shouldn't be required.